sopel_modules/weather/here.py

```diff
@@ -48,7 +51,13 @@ def location(self, search_string):
         if len(results) == 0:
             raise Exception(f"No results found for {search_string}")
         elif len(results) > 1:
-            matches = ' '.join([x["Location"]["Address"]["Label"] for x in results])
+            # If all GPS positions match, return last entry
+            positions = [x['position'] for x in results]
+            if all(i == positions[0] for i in positions):
+                return results[-1]
+
+            # matches = ' '.join([x["Location"]["Address"]["Label"] for x in results])
+            matches = ' '.join([x["address"]["label"] for x in results])
             raise Exception(f"Multiple results found for {search_string}: {matches}")
```

Comment:
matches = dict([(x['position'], x) for x in results])
if len(matches) == 1:
  return matches[0]
raise Exception(f"Multiple results found for {search_string}: {' '.join([x["address"]["label"] for x in matches.values()])}